app/controllers/programmes_controller.rb

```diff
@@ -14,9 +14,16 @@ def show
 
   def sessions
     academic_year = Date.current.academic_year
-    sessions_for_programme = policy_scope(Session).has_programme(@programme)
+    sessions_for_programme =
+      policy_scope(Session)
+        .has_programme(@programme)
+        .strict_loading
+        .includes(:dates, :location)
 
-    @scheduled_sessions = sessions_for_programme.scheduled
+    @scheduled_sessions =
+      sessions_for_programme.scheduled.sort_by do |session|
+        [session.dates.first.value, session.location.name]
```

Comment:
The location can be `nil` if we import a session for an unknown school so I think we'll want to handle that here.

Reply:
Updated in https://github.com/nhsuk/manage-vaccinations-in-schools/pull/1945/commits/5806f935e4bc7c9a04b2de0fbb18a9f45074909f to handle `location = nil`